### PR TITLE
Issue#9 - Added isNull or undefined validation to missing places  in Checkbox.js

### PR DESCRIPTION
**Pull request by sanjaypradeep**
In continuation with https://github.com/javascriptteacher/primary/pull/10

Addition of basic validation in checkbox.js. Please review this PR, comments are welcome, willing to discuss/learn/share things.

### CODE REVIEW

File: library/checkbox/checkbox.js [hunk 1]
```diff
@@ -1,27 +1,42 @@
+import { isNullOrUndefined } from "util";
+
 export class Checkbox {
     constructor() { }
     // Add event listeners to all checkboxes on the page
     static initialize() {
       console.log("Initializing checkbox UI  (convert all '.checkbox' to a slider UI control)");
       // Attach event listener to the checkbox
-      document.querySelectorAll(".checkbox").forEach(function(obj) {
+      let matchingCheckbox = document.querySelectorAll('checkbox');
```
Comment (alexmuzenhardt) [inline library/checkbox/checkbox.js]: We should query all ".checkbox" classes not "checkbox" tags, right?

File: library/checkbox/checkbox.js [hunk 2]
```diff
@@ -1,27 +1,42 @@
...
-        let f = obj.getAttribute("for");
-        if (f) {
-          obj.addEventListener("click", function(event) {
-            let checkbox = document.getElementById(f);
-            if (checkbox) {
-                let state = checkbox.getAttribute("data-value");
-                state == "off" ? checkbox.setAttribute("data-value", "on") : checkbox.setAttribute("data-value", "off");
-            }
-          });
+      let matchingCheckboxLabel = document.querySelectorAll('checkbox-label');
```
Comment (alexmuzenhardt) [inline library/checkbox/checkbox.js]: Same here: ".checkbox-label", not "checkbox-label", right?

File: library/checkbox/checkbox.js [hunk 3]
```diff
@@ -1,27 +1,42 @@
...
+          let state = checkbox.getAttribute("data-value")
+          if (isNullOrUndefined(state)){
+            throw Error("Checkbox value seems to be undefined or nil");
+          }
+          state === "off" ? checkbox.setAttribute("data-value", "on") :
+              checkbox.setAttribute("data-value", "off");
+        });
       });
     }
 };
```
Comment (alexmuzenhardt) [inline library/checkbox/checkbox.js]: no semicolon needed here.

File: library/checkbox/checkbox.js (hunk 1 above)
Comment (javascriptteacher) [reply library/checkbox/checkbox.js]: Exactly, just add the leading . back in. Good call on null though.

Comment (sanjaypradeep) [reply library/checkbox/checkbox.js]: good catch and thanks @alexmuzenhardt @javascriptteacher 
Tested this in a browser, but got missed in this PR.

File: library/checkbox/checkbox.js (hunk 2 above)
Comment (sanjaypradeep) [reply library/checkbox/checkbox.js]: both comments are addressed and committed new changes.

File: library/checkbox/checkbox.js (hunk 3 above)
Comment (sanjaypradeep) [reply library/checkbox/checkbox.js]: I didn't get you. can you please be specific on which line is it not required?

**Comment by alexmuzenhardt** [comment]
Looks good to me now. @javascriptteacher I would merge it then.

**Pull request closed without merging**